app/controllers/deployment_controller.rb

```diff
@@ -0,0 +1,210 @@
+class DeploymentController < ApplicationController
+  extend ActiveSupport::Concern
+
+  # before_action :check_privileges
+  after_action :cleanup_action
+
+  def new
+    @layout = 'openshift_deployment'
+    if params[:id].nil?
+      @breadcrumbs.clear
+    end
+  end
+
+  def data
+    render :json => {:data => collect_data}
+  end
+
+  def create_new_deployment
+    deployment = Deployment.new
+    auth = create_deployment_auth(params)
+    auth.generate_ansible_entry
+    node_tag = create_or_get_tag("node").name
+    master_tag = create_or_get_tag("master").name
+    deployment_master_tag = create_or_get_tag("deployment_master").name
+    deployment.deployment_type = params["deployment_type"]
+    deployment.deployment_method = params["deployment_method"]
+    deployment.ssh_private = params["ssh_private_key"] if params["ssh_private_key"]
+    deployment.ssh_user = params["ssh_user"] ? params["ssh_user"] : "root"
+    if deployment.deployment_method.include? "managed_existing"
+      deployment.deployed_on_ext_management_system = ExtManagementSystem.find params["deployed_on_ext_management_system_id"]
+      create_deployment_nodes(deployment, params["nodes_id_or_ip"], node_tag, true)
+      create_deployment_nodes(deployment, params["masters_id_or_ip"], master_tag, true)
+      create_deployment_nodes(deployment, [params["deployment_master_id_or_ip"]], deployment_master_tag, true)
+    elsif deployment.deployment_method.include? "managed_provisioned"
+      Deployment.add_basic_root_template
+      deployment.deployed_on_ext_management_system = ExtManagementSystem.find params["deployed_on_ext_management_system_id"]
+      ssh_private, ssh_public = generate_ssh_keys
+      deployment.ssh_private = ssh_private
+      deployment.ssh_public = ssh_public
+      deployment.rhn_user = params["rhn_user"] if params["rhn_user"]
+      deployment.rhn_pass = params["rhn_pass"] if params["rhn_pass"]
+      deployment.rhn_pool_id = params["rhn_pool_id"] if params["rhn_pool_id"]
+      create_deployment_nodes(deployment, (1..params["num_of_nods"]), node_tag)
+      create_deployment_nodes(deployment, (1..params["num_of_masters"]), master_tag)
+      create_deployment_nodes(deployment, (1..1), deployment_master_tag)
+    else
+      create_deployment_nodes(deployment, params["nodes_id_or_ip"], node_tag, nil, true)
+      create_deployment_nodes(deployment, params["masters_id_or_ip"], master_tag, nil, true)
+      create_deployment_nodes(deployment, [params["deployment_master_id_or_ip"]], deployment_master_tag, nil, true)
+    end
+    deployment.save!
+    create_automation_request(generate_params(deployment, params))
+    render :json => {:data => "good"}
+  end
+
+  private
+
+  def create_or_get_tag(tag)
+    ent = Tag.find_by_name(tag)
+    unless ent
+      ent = Tag.new(:name => tag)
```

Comment:
You're creating a Tag with no matching Classification.
This means you're missing metadata e.g. description, but that's probably OK.
E.g. `tag2human()` is designed to cope with tags whose category and/or entry Classifications don't exist.
In any case, if you want a working example creating Classifications (and implicitly matching Tags), see https://github.com/cben/manageiq/blob/bfea7bd0ebdd6da5abc9a160b6130158499604a4/app/models/ems_refresh/save_inventory_container.rb#L388-L406

What worries me more is that you're creating a 2-level name e.g. `/user/master` instead of customary 3-level `/namespace/category/entry`.  Having recently read through Classification, Tag and ActsAsTaggable I strongly suspect not all code there can deal with 2-level.  Though I don't have concrete examples off-hand.
Anyway, it'd make sense to group the 3 tags under a common prefix.